### PR TITLE
Parameter names and Javadocs for the Loot system

### PR DESCRIPTION
**Pull request by diesieben07**

### CODE REVIEW

**Review by jriwanek — DISMISSED** [review]

File: data/net/minecraft/world/level/storage/loot/BuiltInLootTables.mapping [hunk 1]
```diff
@@ -1,4 +1,5 @@
-CLASS net/minecraft/world/level/storage/loot/BuiltInLootTables
+CLASS net/minecraft/world/level/storage/loot/BuiltInLootTables net/minecraft/world/level/storage/loot/BuiltInLootTables
```
Comment (Lanse505) [inline data/net/minecraft/world/level/storage/loot/BuiltInLootTables.mapping:1]: Why the double identifier...? @sciwhiz12

File: data/net/minecraft/world/level/storage/loot/GsonAdapterFactory.mapping [hunk 2]
```diff
@@ -1,6 +1,44 @@
...
 		ARG 0 registry
 		ARG 1 id
 		ARG 2 name
 		ARG 3 typeFunction
+	CLASS Builder
+		METHOD <init> (Lnet/minecraft/core/Registry;Ljava/lang/String;Ljava/lang/String;Ljava/util/function/Function;)V
+			ARG 1 registry
+			ARG 2 elementName
+			ARG 3 typeKey
+			ARG 4 function
```
Comment (Lanse505) [inline data/net/minecraft/world/level/storage/loot/GsonAdapterFactory.mapping]: ``function`` -> ``typeGetter``

File: data/net/minecraft/world/level/storage/loot/LootContext.mapping [hunk 3]
```diff
@@ -34,11 +66,35 @@ CLASS net/minecraft/world/level/storage/loot/LootContext
 			ARG 1 seed
 		METHOD withOptionalRandomSeed (JLjava/util/Random;)Lnet/minecraft/world/level/storage/loot/LootContext$Builder;
 			ARG 1 seed
+			ARG 3 random
```
Comment (Lanse505) [inline data/net/minecraft/world/level/storage/loot/LootContext.mapping:69]: ``random`` -> ``seed``

File: data/net/minecraft/world/level/storage/loot/entries/EntryGroup.mapping [hunk 4]
```diff
@@ -1,3 +1,10 @@
-CLASS net/minecraft/world/level/storage/loot/entries/EntryGroup
+CLASS net/minecraft/world/level/storage/loot/entries/EntryGroup net/minecraft/world/level/storage/loot/entries/EntryGroup
```
Comment (Lanse505) [inline data/net/minecraft/world/level/storage/loot/entries/EntryGroup.mapping:1]: Same as before, double indentifiers >.> @sciwhiz12

File: data/net/minecraft/world/level/storage/loot/entries/LootItem.mapping [hunk 5]
```diff
@@ -1,4 +1,11 @@
-CLASS net/minecraft/world/level/storage/loot/entries/LootItem
+CLASS net/minecraft/world/level/storage/loot/entries/LootItem net/minecraft/world/level/storage/loot/entries/LootItem
```
Comment (Lanse505) [inline data/net/minecraft/world/level/storage/loot/entries/LootItem.mapping:1]: Same as before, double indentifiers >.> @sciwhiz12

**Review by Lanse505 — CHANGES_REQUESTED** [review]
Looked good overall.

I've left a few comments that needs to be adressed in theory.
Also there is something wierd with your patch where in some mapping classes contains double class headers, I've tagged @sciwhiz12 about this so he can look into it.

Otherwise great work Dies!

**Review by Darkhax — CHANGES_REQUESTED** [review]
Looks good to me, approval pending the double header issue.

**Review by Lanse505 — APPROVED** [review]

**Comment by Lanse505** [comment]
@Darkhax feel free to approve, double headers is not a bug.

**Review by Lanse505 — APPROVED** [review]

**Review by Darkhax — APPROVED** [review]

**Pull request merged**